### ISSUE TITLE
cloudformation userdata fails suddenly

### ISSUE THREAD

**Issue by imperialjoy** [widdix/aws-s3-virusscan#87]
There seems to be a bug in userdata, where ruby isn't properly installed (Failed to install gem: aws-sdk-s3-1.54.0).  This appeared to happen in the last few days, older instances created a couple of weeks ago are running ok.  From /var/log/cloud-init-output.log

Cloud-init v. 19.3-3.amzn2 running 'modules:final' at Thu, 02 Sep 2021 23:46:56 +0000. Up 32.66 seconds.
+ trap '/opt/aws/bin/cfn-signal -e 1 --stack AV-App --resource ScanAutoScalingGroup --region ap-southeast-2' ERR
+ /opt/aws/bin/cfn-init -v --stack AV-App --resource ScanLaunchConfiguration --region ap-southeast-2
Error occurred during build: Failed to install gem: aws-sdk-s3-1.54.0 (return code 1)
++ /opt/aws/bin/cfn-signal -e 1 --stack AV-App --resource ScanAutoScalingGroup --region ap-southeast-2
ValidationError: Stack arn:aws:cloudformation:ap-southeast-2:709451545702:stack/AV-App/10dd95c0-5e85-11ea-b569-06624269e00a is in CREATE_COMPLETE state and cannot be signaled
Sep 02 23:48:08 cloud-init[2771]: util.py[WARNING]: Failed running /var/lib/cloud/instance/scripts/part-001 [1]
Sep 02 23:48:08 cloud-init[2771]: cc_scripts_user.py[WARNING]: Failed to run module scripts-user (scripts in /var/lib/cloud/instance/scripts)
Sep 02 23:48:08 cloud-init[2771]: util.py[WARNING]: Running module scripts-user (<module 'cloudinit.config.cc_scripts_user' from '/usr/lib/python2.7/site-packages/cloudinit/config/cc_scripts_user.pyc'>) failed
Cloud-init v. 19.3-3.amzn2 finished at Thu, 02 Sep 2021 23:48:08 +0000. Datasource DataSourceEc2.  Up 104.83 seconds

**Comment by accessau-adam** [comment]
Had this issue also - it's because AWS updated their RubyGems to require Ruby2.3 or greater.
The 'main' RubyGems are locked to specific versions - but their dependancies are not (aws-sdk-core, aws-sigv4, aws-partition, aws-sdk-kms) and these dependencies now want Ruby2.3

**Comment by GlauberrBatista** [comment]
I ran into that issue as well and I find I way (not ideal btw) to fix and make it work for now.

I enabled the installation of ruby 2.6 from aws extras and try to create the stack, but I got no luck since the gems were not found after the ruby upgrade. I tried some other solutions, like forcing the rubygems upgrade first but no success as well.
Then I observed that the `rubygems` entry in the template was not using the rubygems from ruby 2.6 so I decided to install all the gems using the `command` directive and it worked.

I opened a pull request in my fork in case you are interested but I don't plan to point it to this repo since it's not a good solution. I just made it to fix my issues for now, until I'm able to figure out what's going on.
https://github.com/GlauberrBatista/aws-s3-virusscan/pull/1/files
